lib/seahorn/BvOpSem2RawMemMgr.cc

```diff
@@ -621,56 +642,59 @@ PtrTy RawMemManager::inttoptr(Expr intVal, const Type &intTy,
                               const Type &ptrTy) const {
   uint64_t intTySz = m_sem.sizeInBits(intTy);
   uint64_t ptrTySz = m_sem.sizeInBits(ptrTy);
-  assert(ptrTySz == ptrSzInBits());
+  assert(ptrTySz == ptrSizeInBits());
 
   Expr res = intVal;
   if (ptrTySz > intTySz)
     res = bv::zext(res, ptrTySz);
   else if (ptrTySz < intTySz)
     res = bv::extract(ptrTySz - 1, 0, res);
-  return res;
+  return PtrTy(res);
 }
 
 Expr RawMemManager::ptrUlt(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doUlt(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doUlt(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrUle(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doUle(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doUle(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrSlt(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doSlt(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doSlt(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrSle(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doSle(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doSle(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrUgt(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doUgt(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doUgt(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrUge(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doUge(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doUge(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrSgt(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doSgt(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doSgt(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrSge(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doSge(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doSge(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrEq(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doEq(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doEq(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrNe(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doNe(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doNe(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
 Expr RawMemManager::ptrSub(PtrTy p1, PtrTy p2) const {
-  return m_ctx.alu().doSub(p1, p2, ptrSzInBits());
+  return m_ctx.alu().doSub(p1.toExpr(), p2.toExpr(), ptrSizeInBits());
 }
+
 Expr RawMemManager::getFatData(PtrTy p, unsigned SlotIdx) {
   LOG("opsem", WARN << "getFatData() not implemented");
   return Expr();
 }
-Expr RawMemManager::setFatData(PtrTy p, unsigned SlotIdx, Expr data) {
+
+RawMemManager::PtrTy RawMemManager::setFatData(PtrTy p, unsigned SlotIdx,
+                                               Expr data) {
   LOG("opsem", WARN << "setFatData() not implemented");
-  return Expr();
+  return PtrTy(m_ctx.alu().getFalse());
```

Comment:
is this even type correct? Used to return null wrapped in an `Expr`, but now returns a potentially legal expression...

Reply:
Fixed.